### PR TITLE
feat(rome_cli): print number of errors in summary

### LINKED ISSUE #3588: 📎 Print the number of files with errors in `check` summary
### Description

^

### PR DESCRIPTION
**Pull request by SuperchupuDev**
## Summary

Prints the number of errors emitted on the `check` summary, or at least tries to. I'm not really experienced in Rust so it's very possible this breaks something / doesn't work as intended, but it does its job, I guess. Feel free to supersede or suggest changes if you have a better approach

Closes #3588

## Test Plan

image

Commands ran:
```sh-session
cargo build
cargo run --bin rome check ./my-cool-ts-project
```
where `my-cool-ts-project` is a directory with a TypeScript file

### CODE REVIEW

File: crates/rome_cli/src/traversal.rs [hunk 1]
```diff
@@ -182,8 +182,8 @@ pub(crate) fn traverse(execution: Execution, mut session: CliSession) -> Result<
     }
 
     // Processing emitted error diagnostics, exit with a non-zero code
-    if has_errors {
-        Err(Termination::CheckError)
+    if errors > 0 {
+        Err(Termination::CheckError(errors))
```
Comment (MichaReiser) [inline crates/rome_cli/src/traversal.rs:186]: Printing the number of errors in the termination is possible but it's more limited in what formatting it supports. 

I would recommend adding the message in the code on lines 131 - 175. That would further allow us to combine the message to `Checked {count} files in {duration}. Found {x} errors`

Comment (SuperchupuDev) [reply crates/rome_cli/src/traversal.rs:186]: Alright, what should I do to CheckError? Do I make it the same as before this PR?

Comment (MichaReiser) [reply crates/rome_cli/src/traversal.rs:186]: If the count is no longer needed, then yes.

**Comment by SuperchupuDev** [comment]
somehow the snapshot tests are failing after updating them, any idea how i can fix it?

File: crates/rome_cli/src/traversal.rs [hunk 2]
```diff
@@ -133,27 +133,37 @@ pub(crate) fn traverse(execution: Execution, mut session: CliSession) -> Result<
...
             TraversalMode::CI { .. } => {
                 console.log(markup! {
-                    <Info>"Checked "{count}" files in "{duration}</Info>
+                    <Info>"Checked "{count}" file(s) in "{duration}</Info>
                 });
+                if errors > 0 {
+                    console.log(markup! {
+                        <Error>"Found "{errors}" error(s)"</Error>
+                    });
+                }
```
Comment (MichaReiser) [inline crates/rome_cli/src/traversal.rs:157]: You can avoid the repetition by creating a new `CheckResult` struct that implements `Display` 

```
console.log(markup!{ CheckResult { count, duration, errors } })
```

**Comment by ematipico** [comment]
> somehow the snapshot tests are failing after updating them, any idea how i can fix it?

We use [`insta`](https://insta.rs/). You have to install it via `cargo`: `cargo install cargo-insta`. Then run `cargo insta review` or `cargo insta accept`

File: crates/rome_cli/tests/snapshots/main_commands_check/apply_noop.snap [hunk 3]
```diff
@@ -18,4 +18,8 @@ If you wish to apply the suggested fixes, use the command rome check --apply-sug
 
 ```
 
+```block
+Fixed 0 file(s) in 413ms
```
[the remark this answers is not in the document]
Comment (SuperchupuDev) [reply crates/rome_cli/tests/snapshots/main_commands_check/apply_noop.snap]: fixed, had to update a check
image

**Review by MichaReiser — APPROVED** [review]

**Review by ematipico — APPROVED** [review]
Thank you! Great contribution this one! 🚀

**Pull request merged**